### ISSUE TITLE
Blueprint is ignored

### ISSUE THREAD

**Issue by flaker** [mitsuhiko/flask#1426]
Hi,

I was trying to create a small sample for another issue and run into a problem. I have a very simple demo app. This is the main file:

```python
from flask import Flask

app = Flask(__name__)

if __name__ == '__main__':
    from sign import sign as sign_blueprint
    app.register_blueprint(sign_blueprint, url_prefix='/sign')
    app.run()
```

Blueprint `sign` lives inside the `sign` directory and the `__init__.py` file there declares it.

```python
from flask import Blueprint
sign = Blueprint('sign', __name__)
from . import views
```

`views` just contains a simple handler that returns the string ok when requested. The problem is that my request was completely ignored. I started debugging and the routes collection has only the `static` route. The details for my blueprint disappeared.

![Structure](https://cloud.githubusercontent.com/assets/627517/7091648/a4491eb6-df81-11e4-9f54-03cf586848c4.png)

Writing 

* http://127.0.0.1:5000/sign/order/2345
* http://127.0.0.1:5000/sign/order/
* http://127.0.0.1:5000/sign/
* http://127.0.0.1:5000/sign

gets the same 404 error. It seems that somehow my blueprint is ignored and no route is created.

Any ideas?

Thank you,

Federico

**Comment by untitaker** [comment]
Please don't use the issue tracker for support.

From what I see, you didn't register your views on your blueprint. Does that solve your issue?

**Comment by flaker** [comment]
Hello,

It is not my intention to use this as s support board (code does not seem very production like, does it?). I understand that importing views (`from . import views`) will subscribe the routes under that package using the `route` decorator and thought I had found an unexpected behaviour. In any case, thanks to your comment I rechecked the views registered and realised that I was using the main app route instead of the blueprint's.

Thank you,

Federico
